go/libraries/doltcore/env/dolt_docs.go

```diff
@@ -100,3 +103,145 @@ func hasDocFile(fs filesys.ReadWriteFS, file string) bool {
 	exists, isDir := fs.Exists(getDocFile(file))
 	return exists && !isDir
 }
+
+// WorkingRootWithDocs returns a copy of the working root that has been updated with the Dolt docs from the file system.
+func WorkingRootWithDocs(ctx context.Context, dbData DbData) (*doltdb.RootValue, error) {
```

Comment:
You've moved these out of environment.go (good) but there are still waaaay too many of them (bad).

There should be only one: UpdateRootWithDocs. It should take a root value and update it with whatever docs are on the file system, then return it.